objects/src/notes/vault.rs

```diff
@@ -0,0 +1,73 @@
+use super::{Asset, Digest, Hasher, NoteError, Vec, Word, WORD_SIZE};
+
+// NOTE VAULT
+// ================================================================================================
+/// An asset container for a note.
+///
+/// A note vault can contain up to 1000 assets. The entire vault can be reduced to a single hash
+/// which is computed by sequentially hashing the list of the vault's assets.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct NoteVault {
+    assets: Vec<Asset>,
+    hash: Digest,
+}
+
+impl NoteVault {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+    /// The maximum number of assets which can be carried by a single note.
+    pub const MAX_NUM_ASSETS: usize = 1000;
```

Comment:
nit for the future: maybe add a comment with context on why this number is `1000`

Reply:
There isn't really a good reasoning as to why 1000. It probably shouldn't be 10 and it shouldn't be 1 million - but the choice between 100, 1000, and 10000 is somewhat arbitrary.